### PR TITLE
List relationship property

### PR DESCRIPTION
**Pull request by brijeshnec**
ready to merge

### CODE REVIEW

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java [hunk 1]
```diff
@@ -40,11 +42,35 @@ class NGSIObject {
...
+	}
+
 	private boolean atContextRequired = false;
 
 	private boolean isVocabProperty = false;
 	private boolean hasVocab = false;
 
+	private boolean isListProperty = false;
+	private boolean hasList = false;
+
```
Comment (amitrnec) [inline Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:73]: Remove unused space

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java [hunk 2]
```diff
@@ -630,11 +676,11 @@ private void validateAttribute(int payloadType, String expandedProperty, String
 			validateArray();
 		} else {
 			if (isLdKeyWord && parent == null && !isProperty && !isRelationship && !isGeoProperty && !isDateTime
-					&& !isLanguageProperty && !isVocabProperty) {
+					&& !isLanguageProperty && !isVocabProperty && !isListProperty && !isListRelationship) {
```
Comment (amitrnec) [inline Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:679]: Please validate this condition

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 3]
```diff
@@ -189,10 +189,18 @@ private static void addProperty(Map<String, Object> resultMap, String key, Objec
...
+			propertyMap.put(NGSIConstants.OBJECT_LIST,
+					((List<Map<String, Object>>) newValue).get(0).get(NGSIConstants.NGSI_LD_HAS_OBJECT_LIST));
+		}
 		if (((List<Map<String, Object>>) newValue).get(0).containsKey(NGSIConstants.NGSI_LD_HAS_VOCAB)) {
 			propertyMap.put(NGSIConstants.VOCAB,
 					((List<Map<String, Object>>) newValue).get(0).get(NGSIConstants.NGSI_LD_HAS_VOCAB));
 		}
+		if (((List<Map<String, Object>>) newValue).get(0).containsKey(NGSIConstants.NGSI_LD_HAS_LIST)) {
+					propertyMap.put(NGSIConstants.LIST,
+						((List<Map<String, Object>>) newValue).get(0).get(NGSIConstants.NGSI_LD_HAS_LIST));
```
Comment (amitrnec) [inline SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:202]: Please implement concise get functionality.

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 4]
```diff
@@ -189,10 +189,18 @@ private static void addProperty(Map<String, Object> resultMap, String key, Objec
...
+		}
```
Comment (amitrnec) [inline SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:203]: Implement concise create also.

File: Commons/src/main/java/com/github/jsonldjava/core/JsonLdApi.java [hunk 5]
```diff
@@ -377,6 +389,38 @@ else if (isRelationship) {
 							continue;
 						}
 					}
+					else if (isListRelationship) {
+//						expanded value for ListRelationship
+//						"https://uri.etsi.org/ngsi-ld/hasObjectList": [
+//						{
+//							"@list": [
+//							{
+//								"@value": "urn:ngsi-ld:Person:Alice"
```
Comment (amitrnec) [inline Commons/src/main/java/com/github/jsonldjava/core/JsonLdApi.java:398]: Remove commented code.

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java [hunk 6]
```diff
@@ -653,6 +703,14 @@ private void validateAttribute(int payloadType, String expandedProperty, String
 			}
 			if (isProperty && hasObject) {
 				throw new ResponseException(ErrorType.BadRequestData, "You can't have properties with an object");
+			}if ((isListRelationship && !hasListObject)) {
+				throw new ResponseException(ErrorType.BadRequestData, "You can't have list relationships without an object");
+			}
+			if ((isListRelationship && hasValue)) {
+				throw new ResponseException(ErrorType.BadRequestData, "You can't have list relationships with a value");
+			}
+			if (isListProperty && hasListObject) {
```
Comment (amitrnec) [inline Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:712]: Validate this exception.

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 7]
```diff
@@ -207,10 +215,18 @@ private static void addProperty(Map<String, Object> resultMap, String key, Objec
...
 			}
+			if (((List<Map<String, Object>>) oldValue).get(0).containsKey(NGSIConstants.NGSI_LD_HAS_OBJECT_LIST)) {
+				propertyMap.put(NGSIConstants.PREVIOUS_OJBECT_LIST,
+						((List<Map<String, Object>>) oldValue).get(0).get(NGSIConstants.NGSI_LD_HAS_OBJECT_LIST));
+			}
 			if (((List<Map<String, Object>>) oldValue).get(0).containsKey(NGSIConstants.NGSI_LD_HAS_VOCAB)) {
 				propertyMap.put(NGSIConstants.PREVIOUS_VOCAB,
 						((List<Map<String, Object>>) oldValue).get(0).get(NGSIConstants.NGSI_LD_HAS_VOCAB));
 			}
+			if (((List<Map<String, Object>>) oldValue).get(0).containsKey(NGSIConstants.NGSI_LD_HAS_LIST)) {
```
Comment (amitrnec) [inline SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:226]: Verify this condition.

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java (hunk 1 above)
Comment (brijeshnec) [reply Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:73]: done

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java (hunk 2 above)
Comment (brijeshnec) [reply Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:679]: done

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java (hunk 7 above)
Comment (brijeshnec) [reply SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:226]: done

File: Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java (hunk 6 above)
Comment (brijeshnec) [reply Commons/src/main/java/com/github/jsonldjava/core/NGSIObject.java:712]: done

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java (hunk 4 above)
Comment (brijeshnec) [reply SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:203]: done

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java (hunk 3 above)
Comment (brijeshnec) [reply SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:202]: done

**Pull request merged**